### PR TITLE
add guide for github actions

### PR DESCRIPTION
**Pull request by dlawin**

### CODE REVIEW

**Review by kylemcnair — APPROVED** [review]

File: docs/guides/ci_guides/dbt_core/github_actions.md [hunk 1]
```diff
@@ -1,4 +1,228 @@
...
+* [Advanced Config](github_actions.md#advanced-config)
+    * [Pull Request Job](github_actions.md#advanced-pull-request-job)
+
+## Basic Config
+
+### Production Job
+
+This job runs in two scenarios, defined in the `on:` section:
+1. When commits are pushed to the main branch
+    * Why?
```
Comment (dlawin) [inline docs/guides/ci_guides/dbt_core/github_actions.md:18]: idk if there's a better way to put this, but trying to explain why the 3 CI triggers are needed (scheduled, prod deploy, pr)

**Comment by dlawin** [comment]
> 



> Would this benefit from quick blurb describing Github Actions?
> 
> We could probably borrow language from the GH docs
> 
> https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#overview

Agree some context would be helpful

File: docs/guides/ci_guides/dbt_core/github_actions.md (hunk 1 above)
Comment (kylemcnair) [reply docs/guides/ci_guides/dbt_core/github_actions.md:18]: Maybe speaking in terms of how this fits into the data warehouse? Trying to think of "why" in terms of the consumers (analysts, business)

Scheduled
* DWH daily refresh with new data
Prod deploy
* push model changes into the DWH
PR
* understand and QA changes before pushing to DWH

Comment (dlawin) [reply docs/guides/ci_guides/dbt_core/github_actions.md]: Reworked the when/whys, let me know how that looks now

**Pull request closed without merging**